### PR TITLE
fix(infotip) : close on outside click

### PR DESCRIPTION
**Pull request by divyanshiGupta**
Fixes https://github.com/openshiftio/openshift.io/issues/3359

Approaches that I have tried : 

- I tried using built in ngx-bootstrap outside click directive but it didnt work.
- I wrote outside click directive and tried using it, it also didnt work. The implementation using the directive can be seen in the first commit.
- Currently I am explicitly hiding popover by adding HostListener and it closes all the popovers that are opened, but this has a problem as multiple infotips can be opened simultaneously as due to `stopPropagation()` of click on infotip icon, the target is never the infotip icon element.

Update:

- The first approach that I tried was not working because fabric8-ui was using 1.8.0 version of `ngx-bootstrap` and outsideClick functionality was not implemented in 1.8.0, version updated to 1.9.3 in this PR - https://github.com/fabric8-ui/fabric8-ui/pull/2947 and it supports outsideClick functionality.
- Added close infotip on scroll event as discussed with @SMahil 
- Since infotip element is the child of the group type link thus the click on infotip icon is propagated to the link, to avoid this behavior I tried the following things :- 
  
  - Used `stopPropagation()` to stop the click from propagating to the parent but if this is used then the `outsideClick` property works as long as the click is not on another infotip icon if one popover is already open.
  -  If I keep the infotip element outside of the link it appears below the link and to resolve this a lot of css needs to be changed as discussed with @SMahil.
  -  I tried to handle the click on the link explicitly by checking if the click is coming from infotip, in which case navigation will not take place, or from the link itself. But this is a hack. 
@sanbornsen @sahil143 any inputs on this?

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @divyanshiGupta for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-ui/fabric8-planner/6f0cc7d5c24e02ac2083273586f608d8bc0256e9/test-keeper.yml).

**Comment by divyanshiGupta** [comment]
@sanbornsen @sahil143 any suggestions or ideas?

**Comment by divyanshiGupta** [comment]
@sanbornsen @sahil143 ^^

**Comment by sanbornsen** [comment]
@sahil143 are we not using the directives for this one?

**Comment by divyanshiGupta** [comment]
@sanbornsen please refer this https://github.com/fabric8-ui/fabric8-planner/pull/2615
@sahil143 mentioned in the above PR that it is only for Select-Dropdown and table-config dropdown.

**Comment by sahil143** [comment]
@sanbornsen @divyanshiGupta I have not tried the directive for the info tips because Bootstrap Popover has a property `outsideClick` to close the popover. Nevertheless, @divyanshiGupta  can you try out the directive. It's in master now.

**Comment by divyanshiGupta** [comment]
@sahil143 in the description of the PR I have mentioned that I tried using the `outsideClick` inbuilt directive property but it didn't work.
Anyways I will try using the directive.

**Comment by divyanshiGupta** [comment]
@sahil143 I tried using the directive but there is the same issue as I mentioned in the 3rd point of the PR description. Due to stopPropagation, the directive is not getting the element reference of the infotip icon.

**Comment by divyanshiGupta** [comment]
@sahil143 l ^^

File: src/app/components_ngrx/planner-list/planner-list.component.html [hunk 1]
```diff
@@ -20,7 +20,7 @@ <h4>Please wait, we are loading your data.</h4>
 </div>
 
 <ng-template #sidePanel>
-  <aside class="f8-sidepanel">
+  <aside class="f8-sidepanel" (scroll)="closePopover()">
```
Comment (sanbornsen) [inline src/app/components_ngrx/planner-list/planner-list.component.html]: Why there is a scroll event here? There is one in the child section. Besides, why do you need to close the popovers on a scroll event?

File: src/app/components_ngrx/planner-list/planner-list.component.ts [hunk 2]
```diff
@@ -533,6 +533,13 @@ export class PlannerListComponent implements OnInit, OnDestroy, AfterViewChecked
     this.store.dispatch(new WorkItemActions.Reoder(payload));
   }
 
+  closePopover() {
```
Comment (sanbornsen) [inline src/app/components_ngrx/planner-list/planner-list.component.ts]: What exactly does this function do? Should have a comment for this one.

File: src/app/components_ngrx/planner-list/planner-list.component.html (hunk 1 above)
Comment (divyanshiGupta) [reply src/app/components_ngrx/planner-list/planner-list.component.html]: As discussed with @SMahil the popovers should be closed on scroll as it is a separate event like outside click. 
@SMahil ^^

File: src/app/components_ngrx/planner-list/planner-list.component.ts (hunk 2 above)
Comment (divyanshiGupta) [reply src/app/components_ngrx/planner-list/planner-list.component.ts]: It closes the popover on scroll. I will add the comment.

File: src/app/components_ngrx/planner-list/planner-list.component.html (hunk 1 above)
Comment (divyanshiGupta) [reply src/app/components_ngrx/planner-list/planner-list.component.html]: @sanbornsen one scroll event is for the side panel and another for the workitem list view

**Comment by divyanshiGupta** [comment]
@rgarg1 are you talking about infotip-icon or infotip popover ?

**Comment by divyanshiGupta** [comment]
@rgarg1  done

**Comment by rgarg1** [comment]
LGTM @divyanshiGupta

**Comment by divyanshiGupta** [comment]
@nimishamukherjee can we merge this ?

**Comment by nimishamukherjee** [comment]
@divyanshiGupta this PR needs approval from @sanbornsen and @sahil143

**Pull request merged**